src/utils/yaml.ts

```diff
@@ -10,3 +11,16 @@ export function generateRandomSubdomain(): string {
 
     return name;
 }
+
+type FieldName = "name";
+
+export function isUnique(units: YamlCron[] | YamlFunction[], fieldName: FieldName): boolean {
+    const unitMap = new Map<string, string>();
+    for (const unit of units || []) {
+        if (unitMap.has(unit[fieldName])) {
+            return false;
+        }
+        unitMap.set(unit[fieldName], unit[fieldName]);
+    }
+    return true;
+}
```

Comment:
This method should be more general and flexible - ideally we would want to call it:
```
isUnique(anyZodArray, anyFieldName);
```

To achieve this you can change the following:
- use generic types instead of FieldName, YamlCron, YamlFunction. This should do the trick:
```
export function isUnique<T>(array: T[] | undefined, fieldName: keyof T): boolean 
```

- I think a Set instead of a Map makes more sense in this case (it's a tad more optimal because we will add a single value and the Set will automatically reject duplicate values)
```
export function isUnique<T>(units: T[] | undefined, field: keyof T): boolean {
    const unitSet = new Set<T[keyof T]>();
    ...
   unitSet.add(unit[fieldName]);
   ...
}
```